fapolicy_analyzer/ui/main_window.py

```diff
@@ -381,29 +383,12 @@ def on_rulesAdminMenu_activate(self, *args, **kwargs):
         self.__set_trustDbMenu_sensitive(True)
 
     def on_profileExecMenu_activate(self, *args):
-        logging.debug("MainWindow::on_profileExecMenu_activate()")
-        dlgProfTest = ProfileDialog()
-        response = dlgProfTest.get_ref().run()
-
-        if response == Gtk.ResponseType.OK:
-            profiling_args = dlgProfTest.get_text()
-            logging.debug(f"Entry text = {profiling_args}")
-            self._fapd_profiler.start_prof_session(profiling_args)
-            fapd_prof_stderr = self._fapd_profiler.fapd_prof_stderr
-            logging.debug(f"Started prof session, stderr={fapd_prof_stderr}")
-
-            # Catch termination event here somehow. Monitor pid? Pass in
-            # object returned from start_prof_session() above, or maintain only
-            # in the faprofiler instance.
-            sleep(4)
-            self._fapd_profiler.stop_prof_session()
-
-        # Add termination and/or close button
-        dlgProfTest.get_ref().destroy()
-
-        self.__pack_main_content(
-            router(ANALYZER_SELECTION.ANALYZE_FROM_AUDIT, fapd_prof_stderr)
-        )
+        page = router(ANALYZER_SELECTION.PROFILER)
+        width = self.get_object("mainWindow").get_size()[0]
+        page.get_object("dirEntry").set_property("width_request", int(width * Sizing.PROFILER_TEXT_ENTRY))
```

Comment:
It would be nice to handle this sizing internally in the `ProfilerPage` class. There might be a signal you can use that fires once the widget is packed into the content area where you can resize it. Having this code in `main_window` just seems out of place.